From 3d6af6853f8692da5a6c59f171cc261a22b82308 Mon Sep 17 00:00:00 2001
From: Angela Czubak <angela.czubak@cavium.com>
Date: Mon, 4 Mar 2019 15:06:12 +0100
Subject: [PATCH 010/386] octeontx2-af: fix rvu_sso_ggrp_taq_flush

ADD_WORK should refer to slot and not lf.
BAR2 aliasing assumes that we chose which PFVF to refer to and
the unit within it we should work on is a slot.

Change-Id: Ib9d9088e30a74a683deb307e8631a0e40c2a38fb
Signed-off-by: Angela Czubak <aczubak@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/5133
Tested-by: sa_ip-sw-jenkins
Reviewed-by: Stanislaw Kardach <Stanislaw.Kardach@cavium.com>
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
index 10dbcf23e4a0..a21c3a09eda3 100644
--- a/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
+++ b/drivers/net/ethernet/marvell/octeontx2/af/rvu_sso.c
@@ -188,7 +188,7 @@ static void rvu_sso_ggrp_taq_flush(struct rvu *rvu, u16 pcifunc, int lf,
 	reg = rvu_read64(rvu, blkaddr, SSO_AF_HWGRPX_TAQ_THR(lf));
 	while ((reg >> 48) & 0x7FF) {
 		rvu_write64(rvu, blkaddr,
-			    SSO_AF_BAR2_ALIASX(lf, SSO_LF_GGRP_OP_ADD_WORK1),
+			    SSO_AF_BAR2_ALIASX(slot, SSO_LF_GGRP_OP_ADD_WORK1),
 			    0x1 << 3);
 get_work:
 		rvu_write64(rvu, ssow_blkaddr,
-- 
2.17.1

